src/test/scala/com/agar/core/region/RegionTest.scala

```diff
@@ -23,25 +27,243 @@ class RegionTest(_system: ActorSystem)
   "A Region Actor" should {
 
     "initialize a fresh region with fresh entities" in {
-      implicit val context: AgarSystem = new AgarSystem {
-        override val timeout: FiniteDuration = 2 seconds
-      }
-
-      val testProbe = TestProbe()
-      val tracer = system.actorOf(Props(new Tracer(testProbe.ref)))
+      implicit val context: AgarSystem = () => 2 seconds
 
-      val region = system.actorOf(Region.props(ActorRef.noSender, tracer, 7680, 4320), "region")
+      val journal = TestProbe()
+      val bridge = TestProbe()
+      val region = system.actorOf(Region.props(7680, 4320, 0)(journal.ref, bridge.ref))
 
       region ! InitRegion(2, 2)
 
-      val expectedCorrectInit = testProbe.expectMsgPF() {
+      val expectedCorrectInit = journal.expectMsgPF() {
         case Initialized(players, energies) => {
           players.size === 2 && energies.size === 2
         }
       }
 
       expectedCorrectInit should be(true)
     }
+
+    "initialize a fresh region with one fresh player in the frontier" in {
+      implicit val context: AgarSystem = () => 2 seconds
+
+      val journal = TestProbe()
+      val bridge = TestProbe()
+      val region = system.actorOf(Region.props(10, 10, 10)(journal.ref, bridge.ref))
+
+      region ! InitRegion(1, 0)
+
+      val expectedCorrectInit = journal.expectMsgPF() {
+        case Initialized(players, energies) => {
+          players.size === 1 && energies.size === 0
+        }
+      }
+
+      expectedCorrectInit should be(true)
+
+      bridge.expectMsgPF(500 millis) {
+        case Virtual(RegisterPlayer(_, _)) => true
+      }
+
+    }
+  }
+
+  "move a player in the frontier" in {
```

Comment:
I think we have a missing test: A player enter in the region
This test should check if a player is `create | virtual`

Reply:
This is covered by "move a player out the frontier and the region"

Reply:
Okay my bad